src/amo/components/Category/index.js

```diff
@@ -56,48 +65,31 @@ export function mapStateToProps(state, ownProps) {
   const filters = {
     addonType: apiAddonType(ownProps.params.visibleAddonType),
     category: ownProps.params.slug,
-    clientApp: ownProps.params.application,
+    page: parsePage(ownProps.location.query.page),
   };
   const pathname = `/${ownProps.params.visibleAddonType}/${filters.category}/`;
-  const queryParams = { page: parsePage(ownProps.location.query.page) };
+  const paginationQueryParams = { page: filters.page };
 
   const category = getCategoryFromState({
     addonType: filters.addonType,
     categorySlug: filters.category,
-    clientApp: filters.clientApp,
+    clientApp: state.api.clientApp,
```

Comment:
Sure thing. I'll see if a quick follow-up patch for converting would work, I think `shallowUntilTarget()` wasn't available when I started the patch 😄